elytron/ELY-1778-enhanced-audit-logging-feature-parity-with-legacy.adoc

```diff
@@ -0,0 +1,124 @@
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/ELY-1778[ELY-1778]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1230[EAP7-1230]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* TBD
+
+=== Testing By
+
+[X] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+* Security
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+* There should be the ability to specify a Java class should be logged by using an @Audit Java annotation, similarly to what was https://developer.jboss.org/wiki/PicketBoxSecurityAnnotations#jive_content_id_SecurityAudit_Annotation[available in Picketbox]
+** This annotation would be available to be added onto a class for signifying when that class is called it should be logged
+
+* Additional ```AuditEvent``` should be added for feature parity with Legacy Security Audit Logging. The following table lists a legacy event, its use case, and what its name will be in WildFly Elytron:
+|===
+|Legacy Event | Use Case | Elytron Name
+
+| PreAuthenticateEvent
+| Creates a log entry just before authentication begins
+| PreAuthenticateEvent
+
+| PostAuthenticateEvent
+| Creates a log entry just after authentication begins
+| PostAuthenticateEvent
+
+| PreLoggedOutEvent
+| Creates a log entry just before the user un-authenticates
+| PreLoggedOutEvent
+
+| PostLoggedOutEvent
+| Creates a log entry just after the user un-authenticates
+| PostLoggedOutEvent
+
+| Audit Log Enabled
+| Creates a log entry for the audit logging being enabled
+| AuditLogEnabledEvent
+
+| Audit Log Disabled
+| Creates a log entry for the audit logging being disabled
+| AuditLogDisabledEvent
+
+|===
+
+* The following table shows legacy events that are already handled by an Elytron equivalent:
+|===
+|Event |Use |Handled By
+
+| LoggedInEvent
+| Event for when a user successfully logs in
+| SecurityAuthenticationSuccessfulEvent
+
+| AlreadyLoggedInEvent
+| Fires when an already authenticated users attempts to authenticate again
+| SecurityAuthenticationFailedEvent
+
+| LoginFailedEvent
+| Fires when an authentication attempt fails
+| SecurityAuthenticationFailedEvent
+
+| LockedAccountEvent
+| Fires during the authentication process if the account is disabled
+| SecurityAuthenticationFailedEvent
+
+|===
+
+* Additional ```SecurityEvent``` that should be added as ```AuditEvent``` that do not have a direct Legacy equivalent are:
+** RealmAbandonedAuthenticationEvent
+** RealmAuthenticationEvent
+** RealmAuthorizationEvent
+** RealmDefiniteOutcomeAuthenticationEvent
+** RealmFailedAuthenticationEvent
+** RealmIdentityAuthorizationEvent
+** RealmIdentityFailedauthorizationEvent
+** RealmIdentitySuccessfulAuthorizationEvent
+** RealmSuccessfulAuthenticationEvent
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
+
+== Test Plan
+
+Tests for the new Audit Logging capabilities will be added within WildFly Elytron.
```

Comment:
Specified that unit tests will be added. The tests will just be in Elytron since, as far as I've been able to tell, the code will only be within Elytron and not require any code in WildFly or WildFly Core. I will update this section if I find that code within the Elytron subsystem in WildFly Core is required